[Camel 3.5/3.4.3] xml-io should pass namespace info to NamespaceAware elements

Hi I'm trying to use xml based route using routes. I've a xpath expression which uses namespace and it's not taking defined namespaces in routes xml. How do I define namespaces? Here is my route xml

```xml
<routes xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xmlns="http://camel.apache.org/schema/spring"
        xmlns:soap-env="http://www.w3.org/2003/05/soap-envelope"
        xsi:schemaLocation="
            http://camel.apache.org/schema/spring
            http://camel.apache.org/schema/spring/camel-spring-3.4.2.xsd">
    <route id="myroute">
        <from uri="platform-http:/handler?httpMethodRestrict=POST"/>
        <setProperty name="MessageType">
            <xpath resultType="java.lang.String" logNamespaces="true">
                local-name(/soap-env:Envelope/soap-env:Body/*[1])
            </xpath>
        </setProperty>
        <log message="${exchangeProperty.MessageType}"/>
        <setBody>
            <simple>
                Hello ${body}
            </simple>
        </setBody>
    </route>
</routes>
```

When I try to post XML file to handler endpoint I get the following error:

```
org.apache.camel.language.xpath.InvalidXPathException: Invalid xpath: local-name(/soap-env:Envelope/soap-env:Body/*[1]). Reason: javax.xml.xpath.XPathExpressionException: javax.xml.transform.TransformerException: Prefix must resolve to a namespace: soap-env
    at org.apache.camel.language.xpath.XPathBuilder.evaluateAs(XPathBuilder.java:943)
    at org.apache.camel.language.xpath.XPathBuilder.evaluate(XPathBuilder.java:921)
    at org.apache.camel.language.xpath.XPathBuilder.evaluate(XPathBuilder.java:240)
    at org.apache.camel.processor.SetPropertyProcessor.process(SetPropertyProcessor.java:47)
    at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.run(RedeliveryErrorHandler.java:395)
    at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.schedule(DefaultReactiveExecutor.java:148)
    at org.apache.camel.impl.engine.DefaultReactiveExecutor.scheduleMain(DefaultReactiveExecutor.java:60)
    at org.apache.camel.processor.Pipeline.process(Pipeline.java:147)
    at org.apache.camel.processor.CamelInternalProcessor.process(CamelInternalProcessor.java:286)
    at org.apache.camel.quarkus.component.platform.http.runtime.QuarkusPlatformHttpConsumer.lambda$doStart$2(QuarkusPlatformHttpConsumer.java:123)
.....................
Caused by: javax.xml.xpath.XPathExpressionException: javax.xml.transform.TransformerException: Prefix must resolve to a namespace: soap-env
    at java.xml/com.sun.org.apache.xpath.internal.jaxp.XPathImpl.compile(XPathImpl.java:170)
    at org.apache.camel.language.xpath.XPathBuilder.createXPathExpression(XPathBuilder.java:1170)
    at org.apache.camel.language.xpath.XPathBuilder.evaluateAs(XPathBuilder.java:941)
    ... 29 more
Caused by: javax.xml.transform.TransformerException: Prefix must resolve to a namespace: soap-env
    at java.xml/com.sun.org.apache.xpath.internal.compiler.XPathParser.error(XPathParser.java:621)
    at java.xml/com.sun.org.apache.xpath.internal.compiler.Lexer.mapNSTokens(Lexer.java:637)
    at java.xml/com.sun.org.apache.xpath.internal.compiler.Lexer.tokenize(Lexer.java:269)
```

Can anyone point me how to define custom namespaces?

--- a/integration-tests/main-xml-io/pom.xml
+++ b/integration-tests/main-xml-io/pom.xml
@@ -59,6 +59,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-resteasy-jsonb</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-xpath</artifactId>
+        </dependency>
 
         <!-- test dependencies -->
         <dependency>
@@ -133,6 +137,19 @@
         <dependency>
             <groupId>org.apache.camel.quarkus</groupId>
             <artifactId>camel-quarkus-xml-io-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-xpath-deployment</artifactId>
             <version>${project.version}</version>
             <type>pom</type>
             <scope>test</scope>

--- a/integration-tests/main-xml-io/src/main/java/org/apache/camel/quarkus/main/CoreMainXmlIoResource.java
+++ b/integration-tests/main-xml-io/src/main/java/org/apache/camel/quarkus/main/CoreMainXmlIoResource.java
@@ -22,18 +22,24 @@ import javax.json.Json;
 import javax.json.JsonArrayBuilder;
 import javax.json.JsonObject;
 import javax.json.JsonObjectBuilder;
+import javax.ws.rs.Consumes;
 import javax.ws.rs.GET;
+import javax.ws.rs.POST;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
 
 import org.apache.camel.ExtendedCamelContext;
+import org.apache.camel.ProducerTemplate;
 
 @Path("/test")
 @ApplicationScoped
 public class CoreMainXmlIoResource {
     @Inject
     CamelMain main;
+
+    @Inject
+    ProducerTemplate template;
 
     @Path("/main/describe")
     @GET
@@ -68,5 +74,13 @@ public class CoreMainXmlIoResource {
                 .add("routes", routes)
                 .add("autoConfigurationLogSummary", main.getMainConfigurationProperties().isAutoConfigurationLogSummary())
                 .build();
+    }
+
+    @Path("/xml-io/namespace-aware")
+    @POST
+    @Produces(MediaType.TEXT_PLAIN)
+    @Consumes(MediaType.APPLICATION_XML)
+    public String namespaceAware(String body) {
+        return template.requestBody("direct:namespace-aware", body, String.class);
     }
 }

--- a/integration-tests/main-xml-io/src/main/resources/routes/my-routes.xml
+++ b/integration-tests/main-xml-io/src/main/resources/routes/my-routes.xml
@@ -19,6 +19,7 @@
 -->
 <routes xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xmlns="http://camel.apache.org/schema/spring"
+        xmlns:foo="http://camel.apache.org/foo"
         xsi:schemaLocation="
             http://camel.apache.org/schema/spring
             http://camel.apache.org/schema/spring/camel-spring.xsd">
@@ -29,6 +30,15 @@
             <constant>Hello World!!!</constant>
         </setBody>
         <to uri="log:from-xml"/>
+    </route>
+
+    <route id="namespace-aware-route">
+        <from uri="direct:namespace-aware"/>
+        <setBody>
+           <xpath resultType="java.lang.String">
+              /foo:foo-text/text()
+           </xpath>
+        </setBody>
     </route>
 
 </routes>

--- a/integration-tests/main-xml-io/src/test/java/org/apache/camel/quarkus/main/CoreMainXmlIoTest.java
+++ b/integration-tests/main-xml-io/src/test/java/org/apache/camel/quarkus/main/CoreMainXmlIoTest.java
@@ -20,6 +20,7 @@ import javax.ws.rs.core.MediaType;
 
 import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
 import io.restassured.path.json.JsonPath;
 import org.apache.camel.quarkus.core.DisabledModelJAXBContextFactory;
 import org.apache.camel.quarkus.core.DisabledModelToXMLDumper;
@@ -27,6 +28,7 @@ import org.apache.camel.xml.in.ModelParserXMLRoutesDefinitionLoader;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.core.Is.is;
 
 @QuarkusTest
 public class CoreMainXmlIoTest {
@@ -49,5 +51,20 @@ public class CoreMainXmlIoTest {
                 .isEmpty();
         assertThat(p.getList("routes", String.class))
                 .contains("my-xml-route");
+    }
+
+    @Test
+    public void namespaceAware() {
+        String message = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                + "<foo:foo-text xmlns:foo=\"http://camel.apache.org/foo\">bar</foo:foo-text>";
+
+        RestAssured.given()
+                .contentType(ContentType.XML)
+                .body(message)
+                .post("/test/xml-io/namespace-aware")
+                .then()
+                .statusCode(200)
+                .body(is("bar"));
+
     }
 }

--- a/integration-tests/weather/pom.xml
+++ b/integration-tests/weather/pom.xml
@@ -69,7 +69,7 @@
 
         <!-- The following dependencies guarantee that this module is built after them. You can update them by running `mvn process-resources -Pformat -N` from the source tree root directory -->
 
-        <!-- The following dependencies guarantee that this module is built after them. You can update them by runing `mvn process-resources -Pformat -N` from the source tree root directory -->
+        <!-- The following dependencies guarantee that this module is built after them. You can update them by running `mvn process-resources -Pformat -N` from the source tree root directory -->
         <dependency>
             <groupId>org.apache.camel.quarkus</groupId>
             <artifactId>camel-quarkus-weather-deployment</artifactId>

